nursery/swap.go

```diff
@@ -288,12 +288,16 @@ func (nursery *Nursery) handleSwapStatus(swap *database.Swap, status boltz.SwapS
 			return
 		}
 		invoiceAmount := uint64(decodedInvoice.MilliSat.ToSatoshis())
-		serviceFee := swap.ServiceFeePercent.Calculate(swap.ExpectedAmount)
-		boltzOnchainFee := swap.ExpectedAmount - invoiceAmount - serviceFee
+		serviceFee := swap.ServiceFeePercent.Calculate(invoiceAmount)
+		boltzOnchainFee := int64(swap.ExpectedAmount - invoiceAmount - serviceFee)
+		if boltzOnchainFee < 0 {
+			logger.Warnf("Boltz onchain fee seems to be negative")
```

Comment:
How could that even happen? It shouldn't be possible at all and doing that check after casting the `uint64` result of the calculation to `int64` wouldn't even work I guess?

Reply:
Nevermind; I am wrong